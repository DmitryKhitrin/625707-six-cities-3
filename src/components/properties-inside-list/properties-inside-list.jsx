import React from "react";
import PropTypes from "prop-types";
import {ProppertyInside} from "../property-inside/property-inside.jsx";

export const PropertiesInsideList = ({propertyInside = []}) => {
  return (
    <div className="property__inside">
      <h2 className="property__inside-title">What&apos;s inside</h2>
      <ul className="property__inside-list">
        {propertyInside.map((property, i) => {
          return <ProppertyInside key={i} property={property} />;
        })}
      </ul>
    </div>
  );
};

PropertiesInsideList.propTypes = {
  propertyInside: PropTypes.arrayOf(PropTypes.string).isRequired
};
