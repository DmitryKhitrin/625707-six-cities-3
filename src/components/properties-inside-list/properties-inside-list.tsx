import React, {FC, memo} from "react";
import {ProppertyInside} from "../property-inside/property-inside";

type Props = {
  propertyInside: string[];
}

const PropertiesInsideList: FC<Props> = ({propertyInside = []}) => {
  return (
    <div className="property__inside">
      <h2 className="property__inside-title">What&apos;s inside</h2>
      <ul className="property__inside-list">
        {propertyInside.map((property, i) => {
          return <ProppertyInside key={i} property={property} />;
        })}
      </ul>
    </div>
  );
};

const MemoizedPropertiesInsideList = memo(PropertiesInsideList);
export {MemoizedPropertiesInsideList as PropertiesInsideList};
