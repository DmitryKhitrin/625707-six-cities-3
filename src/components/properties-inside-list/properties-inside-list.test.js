import React from "react";
import renderer from "react-test-renderer";
import {PropertiesInsideList} from "./properties-inside-list.jsx";

const mocData = [`Ванная`, `Камин`, `Стул`, `Стол`, `Ложка`];

it(`<PropertiesInsideList /> should render main.`, () => {
  const tree = renderer
    .create(<PropertiesInsideList propertyInside={mocData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
