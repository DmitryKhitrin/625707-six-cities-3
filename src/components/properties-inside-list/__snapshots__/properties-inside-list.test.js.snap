// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PropertiesInsideList /> should render main. 1`] = `
<div
  className="property__inside"
>
  <h2
    className="property__inside-title"
  >
    What's inside
  </h2>
  <ul
    className="property__inside-list"
  >
    <li
      className="property__inside-item"
    >
      Ванная
    </li>
    <li
      className="property__inside-item"
    >
      Камин
    </li>
    <li
      className="property__inside-item"
    >
      Стул
    </li>
    <li
      className="property__inside-item"
    >
      Стол
    </li>
    <li
      className="property__inside-item"
    >
      Ложка
    </li>
  </ul>
</div>
`;
